src/index.js

```diff
@@ -536,10 +544,21 @@ class KadDHT extends EventEmitter {
 
         each(peers, (peer, cb) => {
           this._log('putProvider %s to %s', key.toBaseEncodedString(), peer.toB58String())
-          this.network.sendMessage(peer, msg, cb)
+          this.network.sendMessage(peer, msg, (err) => {
+            if (err) errors.push(err)
+            cb()
+          })
         }, cb)
       }
-    ], (err) => callback(err))
+    ], (err) => {
+      if (errors.length) {
```

Comment:
This should help mitigate https://github.com/libp2p/js-libp2p-kad-dht/issues/105. While the provide will still technically fail, we'll at least have attempted to provide to all peers.